server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -336,7 +336,11 @@ public SearchShardTarget getShard() {
     }
 
     public void shard(SearchShardTarget target) {
-        if (innerHits != null) {
+        shard(target, true);
+    }
+
+    public void shard(SearchShardTarget target, boolean fillInnerHits) {
```

Comment:
I guess it's not an option to have innerHits set to null instead of having to provide a boolean flag to set them? If not, maybe this method can be made package private?

Reply:
++, I'll change to package private